R/data.table.R

```diff
@@ -2457,6 +2457,14 @@ setattr = function(x,name,value) {
   invisible(x)
 }
 
+setlevels = function(x, value) {
+  if (anyDuplicated(value) == 0) {
```

Comment:
I think we have an internal any_duplicated (or completely different name) which is specialized and faster than base for character strings. If so, this anyDuplicated call should be moved down to C to use the internal one.

Reply:
I completely re-wrote `setlevels` please let me know what you think.

Reply:
Removing the WIP except if you think it needs  further changes?